test/test_qa_utils.py

```diff
@@ -1048,3 +1115,256 @@ def test_mock_keys_not_implemented():
 
     err = MockKeysNotImplemented(keys=['foo', 'bar'], operation="some-operation")
     assert str(err) == 'Mocked some-operation does not implement keywords: foo, bar'
+
+
+class SampleClass:
+
+    def __init__(self, favorite_fruit):
+        self.favorite_fruit = favorite_fruit
+
+    FAVORITE_COLOR = CustomizableProperty('FAVORITE_COLOR', description="the string name of a color")
+
+
+class SampleClass2(SampleClass):
+
+    FAVORITE_SONG = CustomizableProperty('FAVORITE_SONG', description="the string name of a song")
+
+
+class SampleClass3(SampleClass2):
+
+    FAVORITE_COLOR = 'blue'
+    FAVORITE_SONG = 'Jingle Bells'
+
+
+def test_find_field_declared_class():
+
+    thing = SampleClass2(favorite_fruit='orange')
+
+    [kind, value] = UncustomizedInstance._find_field_declared_class(thing, 'favorite_fruit')
+    assert kind == 'instance'
+    assert value == 'orange'
```

Comment:
So use case here is to see which class in a potentially large number of parent classes does a property belong to?

Reply:
Yeah, it tells you whose documentation you need to read.